Apps/Common/src/Services/NonceService.cs

```diff
@@ -0,0 +1,47 @@
+﻿// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+
+namespace HealthGateway.Common.Services
+{
+    using System;
+    using System.Security.Cryptography;
+
+    /// <summary>
+    /// Implementation of the <see cref="NonceService"/> with cryptograpgic generation.
+    /// </summary>
+    public class NonceService : INonceService
+    {
```

Comment:
TBD: This seems more of a delegate than a service based on what we've done in the past.  Instead of Nonce, this could be secure random generation which would have broader user.  
No Unit Tests:  The unit test could at least confirm that two are different and that they have the correct size.

Reply:
Good point, I could make it general. Once it goes into dev I will take a look at how its working and enhance it